src/core/dbus-unit.c

```diff
@@ -864,6 +864,7 @@ const sd_bus_vtable bus_unit_vtable[] = {
         SD_BUS_PROPERTY("OnFailureJobMode", "s", property_get_job_mode, offsetof(Unit, on_failure_job_mode), SD_BUS_VTABLE_PROPERTY_CONST),
         SD_BUS_PROPERTY("IgnoreOnIsolate", "b", bus_property_get_bool, offsetof(Unit, ignore_on_isolate), SD_BUS_VTABLE_PROPERTY_CONST),
         SD_BUS_PROPERTY("NeedDaemonReload", "b", property_get_need_daemon_reload, 0, SD_BUS_VTABLE_PROPERTY_CONST),
+        SD_BUS_WRITABLE_PROPERTY("NeedsRestart", "b", bus_property_get_bool, bus_property_set_bool, offsetof(Unit, needs_restart), 0),
```

Comment:
So I shied away from SetUnitProperties / set-properties because those calls like to store the unit state on disk. But I'll just add a check to avoid that for NeedsRestart/NeedsReload. You are right, SetUnitProperties is a better interface for this.